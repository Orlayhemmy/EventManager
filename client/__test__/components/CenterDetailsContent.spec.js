import React from 'react';
import Connected, { CenterDetailsContent } from '../../components/CenterDetails/Template/Content/CenterDetailsContent';
// import { center, userState } from './defaultProps';

describe('center details component', () => {
  const props = {
    eventState: {
      event: {
        eventTitle: ''
      }
    },
    path: '/',
    getEventSelected: jest.fn(),
    modifyCenterEvent: jest.fn(),
    deleteCenterEvent: jest.fn(),
    getCenterEvents: jest.fn(),
    clearEventState: jest.fn(),
    getCenterSelected: jest.fn(),
    modifyCenter: jest.fn(),
    uploadImage: jest.fn(),
    showDiv: jest.fn(),
    centerData: {
      center: {
        id: 1,
        centerName: '',
        facilities: [],
        location: '',
        capacity: ''
      },
      status: 202
    }
  };
  const event = {
    preventDefault: jest.fn(),
    target: {
      id: 'approve',
      value: 'Epic',
      files: [{ data: 'picture.jpg', type: 'image/jpeg' }],
      result: '',
      parentNode: {
        id: 'approve'
      },
      dataset: {
        toggleId: ''
      }
    }
  };

  const wrapper = shallow(<CenterDetailsContent {...props} />);

  wrapper.setProps({
    centerData: {
      center: {
        centerName: 'Epic Center',
        location: 'Lekki',
        description: 'A center with emotions',
        facilities: 'car park',
        capacity: '500',
        imageUrl: 'picture.jpg',
        cost: '50000',
        image: 'newpicture.jpg'
      }
    }
  });
  it('should render the component based', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('return true when there is no error on isValid call', () => {
    const spy = sinon.spy(wrapper.instance(), 'isValid');

    wrapper.instance().isValid(event);
    expect(spy.calledOnce).toBeTruthy();

    wrapper.setState({
      location: 'Lekki#',
      description: 'A center with emotions',
      facilities: 'car park',
      capacity: '500',
      image: 'picture.jpg',
      cost: '50000'
    });
    wrapper.instance().isValid(event);
    expect(spy.calledTwice).toBeTruthy();
    expect(wrapper.state().errors.location).toEqual('location can not include symbols except comma');
    wrapper.setState({
      ...wrapper.props,
      location: 'Lekki'
    });
    wrapper.instance().isValid(event);
    expect(wrapper.state().location).toEqual('Lekki');
  });
  it('call the onchange function and set state', () => {
    const spy = sinon.spy(wrapper.instance(), 'onChange');
    wrapper.instance().onChange(event);
    expect(spy.calledOnce).toBeTruthy();
    expect(wrapper.state().centerName).toEqual('Epic Center');
  });
  it('call the submit function', () => {
    const spy = sinon.spy(wrapper.instance(), 'onSubmit');
    wrapper.setState({
      ...wrapper.props,
      errors: ''
    });
    wrapper.instance().onSubmit(event);
    expect(spy.calledOnce).toBeTruthy();
    expect(wrapper.state().location).toEqual('Lekki');
    expect(wrapper.state().capacity).toEqual('500');
    expect(props.uploadImage).toHaveBeenCalledTimes(2);
    wrapper.setState({
      ...wrapper.props,
      centerName: 'Andela',
      image: 'picture.jpg'
    });

    // expect(props.modifyCenter).toHaveBeenCalledTimes(1);
  });
  it('call showImage function', () => {
    const spy = sinon.spy(wrapper.instance(), 'showImage');
    wrapper.instance().showImage(event);
    expect(spy.calledOnce).toBeTruthy();
    expect(wrapper.state().image).toEqual('picture.jpg');
  });

  it('call onclick to set state', () => {
    const spy = sinon.spy(wrapper.instance(), 'onClick');
    wrapper.instance().onClick(event);
    expect(spy.calledOnce).toBeTruthy();
    expect(wrapper.state().decision).toEqual('approve');
  });

  it('call onApprove function to dispatch update action', () => {
    const spy = sinon.spy(wrapper.instance(), 'onApprove');
    wrapper.instance().onApprove(event);
    expect(spy.calledOnce).toBeTruthy();
    expect(props.modifyCenterEvent).toHaveBeenCalledTimes(1);
    expect(wrapper.state().decision).toEqual('approve');
  });
});

describe('Connected Component', () => {
  it('should render the connected component with all props', () => {
    const store = mockStore({ centerData: {}, eventState: {} });
    const wrapper = shallow(<Connected store={store} />);
    expect(wrapper.length).toEqual(1);
  });
});
