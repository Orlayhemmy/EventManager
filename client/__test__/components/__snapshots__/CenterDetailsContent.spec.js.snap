// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`center details component should render the component based 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CenterDetailsContent
    centerData={
      Object {
        "center": Object {
          "capacity": "500",
          "centerName": "Epic Center",
          "cost": "50000",
          "description": "A center with emotions",
          "facilities": "car park",
          "image": "newpicture.jpg",
          "imageUrl": "picture.jpg",
          "location": "Lekki",
        },
      }
    }
    clearEventState={[MockFunction]}
    deleteCenterEvent={[MockFunction]}
    eventState={
      Object {
        "event": Object {
          "eventTitle": "",
        },
      }
    }
    getCenterEvents={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    getCenterSelected={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    getEventSelected={[MockFunction]}
    modifyCenter={[MockFunction]}
    modifyCenterEvent={[MockFunction]}
    path="/"
    showDiv={[MockFunction]}
    uploadImage={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <div
          className="row"
        >
          <div
            className="col-lg-6 card bb text-center pt-4 pb-4"
          >
            <Details
              centerState={
                Object {
                  "capacity": "500",
                  "centerName": "Epic Center",
                  "cost": "50000",
                  "decision": "",
                  "description": "A center with emotions",
                  "errors": "",
                  "event": "",
                  "facilities": "car park",
                  "id": "",
                  "image": "newpicture.jpg",
                  "imageUrl": "picture.jpg",
                  "location": "Lekki",
                }
              }
              showHiddenDiv={[Function]}
            />
            <EditCenter
              centerState={
                Object {
                  "capacity": "500",
                  "centerName": "Epic Center",
                  "cost": "50000",
                  "decision": "",
                  "description": "A center with emotions",
                  "errors": "",
                  "event": "",
                  "facilities": "car park",
                  "id": "",
                  "image": "newpicture.jpg",
                  "imageUrl": "picture.jpg",
                  "location": "Lekki",
                }
              }
              imageUrl="newpicture.jpg"
              isValid={[Function]}
              onChange={[Function]}
              path="/"
              showHiddenDiv={[Function]}
              showImage={[Function]}
            />
          </div>
          <EventsList
            eventState={undefined}
            onClick={[Function]}
          />
          <DeleteModal
            content={
              <DeleteContent
                onApprove={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                showHiddenDiv={[Function]}
                state={
                  Object {
                    "capacity": "500",
                    "centerName": "Epic Center",
                    "cost": "50000",
                    "decision": "",
                    "description": "A center with emotions",
                    "errors": "",
                    "event": "",
                    "facilities": "car park",
                    "id": "",
                    "image": "newpicture.jpg",
                    "imageUrl": "picture.jpg",
                    "location": "Lekki",
                  }
                }
              />
            }
            path="/"
          />
        </div>
      </div>,
      "id": "center-event",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className="col-lg-6 card bb text-center pt-4 pb-4"
          >
            <Details
              centerState={
                Object {
                  "capacity": "500",
                  "centerName": "Epic Center",
                  "cost": "50000",
                  "decision": "",
                  "description": "A center with emotions",
                  "errors": "",
                  "event": "",
                  "facilities": "car park",
                  "id": "",
                  "image": "newpicture.jpg",
                  "imageUrl": "picture.jpg",
                  "location": "Lekki",
                }
              }
              showHiddenDiv={[Function]}
            />
            <EditCenter
              centerState={
                Object {
                  "capacity": "500",
                  "centerName": "Epic Center",
                  "cost": "50000",
                  "decision": "",
                  "description": "A center with emotions",
                  "errors": "",
                  "event": "",
                  "facilities": "car park",
                  "id": "",
                  "image": "newpicture.jpg",
                  "imageUrl": "picture.jpg",
                  "location": "Lekki",
                }
              }
              imageUrl="newpicture.jpg"
              isValid={[Function]}
              onChange={[Function]}
              path="/"
              showHiddenDiv={[Function]}
              showImage={[Function]}
            />
          </div>
          <EventsList
            eventState={undefined}
            onClick={[Function]}
          />
          <DeleteModal
            content={
              <DeleteContent
                onApprove={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                showHiddenDiv={[Function]}
                state={
                  Object {
                    "capacity": "500",
                    "centerName": "Epic Center",
                    "cost": "50000",
                    "decision": "",
                    "description": "A center with emotions",
                    "errors": "",
                    "event": "",
                    "facilities": "car park",
                    "id": "",
                    "image": "newpicture.jpg",
                    "imageUrl": "picture.jpg",
                    "location": "Lekki",
                  }
                }
              />
            }
            path="/"
          />
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="col-lg-6 card bb text-center pt-4 pb-4"
            >
              <Details
                centerState={
                  Object {
                    "capacity": "500",
                    "centerName": "Epic Center",
                    "cost": "50000",
                    "decision": "",
                    "description": "A center with emotions",
                    "errors": "",
                    "event": "",
                    "facilities": "car park",
                    "id": "",
                    "image": "newpicture.jpg",
                    "imageUrl": "picture.jpg",
                    "location": "Lekki",
                  }
                }
                showHiddenDiv={[Function]}
              />
              <EditCenter
                centerState={
                  Object {
                    "capacity": "500",
                    "centerName": "Epic Center",
                    "cost": "50000",
                    "decision": "",
                    "description": "A center with emotions",
                    "errors": "",
                    "event": "",
                    "facilities": "car park",
                    "id": "",
                    "image": "newpicture.jpg",
                    "imageUrl": "picture.jpg",
                    "location": "Lekki",
                  }
                }
                imageUrl="newpicture.jpg"
                isValid={[Function]}
                onChange={[Function]}
                path="/"
                showHiddenDiv={[Function]}
                showImage={[Function]}
              />
            </div>,
            <EventsList
              eventState={undefined}
              onClick={[Function]}
            />,
            <DeleteModal
              content={
                <DeleteContent
                  onApprove={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  showHiddenDiv={[Function]}
                  state={
                    Object {
                      "capacity": "500",
                      "centerName": "Epic Center",
                      "cost": "50000",
                      "decision": "",
                      "description": "A center with emotions",
                      "errors": "",
                      "event": "",
                      "facilities": "car park",
                      "id": "",
                      "image": "newpicture.jpg",
                      "imageUrl": "picture.jpg",
                      "location": "Lekki",
                    }
                  }
                />
              }
              path="/"
            />,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Details
                  centerState={
                    Object {
                      "capacity": "500",
                      "centerName": "Epic Center",
                      "cost": "50000",
                      "decision": "",
                      "description": "A center with emotions",
                      "errors": "",
                      "event": "",
                      "facilities": "car park",
                      "id": "",
                      "image": "newpicture.jpg",
                      "imageUrl": "picture.jpg",
                      "location": "Lekki",
                    }
                  }
                  showHiddenDiv={[Function]}
                />,
                <EditCenter
                  centerState={
                    Object {
                      "capacity": "500",
                      "centerName": "Epic Center",
                      "cost": "50000",
                      "decision": "",
                      "description": "A center with emotions",
                      "errors": "",
                      "event": "",
                      "facilities": "car park",
                      "id": "",
                      "image": "newpicture.jpg",
                      "imageUrl": "picture.jpg",
                      "location": "Lekki",
                    }
                  }
                  imageUrl="newpicture.jpg"
                  isValid={[Function]}
                  onChange={[Function]}
                  path="/"
                  showHiddenDiv={[Function]}
                  showImage={[Function]}
                />,
              ],
              "className": "col-lg-6 card bb text-center pt-4 pb-4",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "centerState": Object {
                    "capacity": "500",
                    "centerName": "Epic Center",
                    "cost": "50000",
                    "decision": "",
                    "description": "A center with emotions",
                    "errors": "",
                    "event": "",
                    "facilities": "car park",
                    "id": "",
                    "image": "newpicture.jpg",
                    "imageUrl": "picture.jpg",
                    "location": "Lekki",
                  },
                  "showHiddenDiv": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "centerState": Object {
                    "capacity": "500",
                    "centerName": "Epic Center",
                    "cost": "50000",
                    "decision": "",
                    "description": "A center with emotions",
                    "errors": "",
                    "event": "",
                    "facilities": "car park",
                    "id": "",
                    "image": "newpicture.jpg",
                    "imageUrl": "picture.jpg",
                    "location": "Lekki",
                  },
                  "imageUrl": "newpicture.jpg",
                  "isValid": [Function],
                  "onChange": [Function],
                  "path": "/",
                  "showHiddenDiv": [Function],
                  "showImage": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "eventState": undefined,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "content": <DeleteContent
                onApprove={[Function]}
                onChange={[Function]}
                onClick={[Function]}
                showHiddenDiv={[Function]}
                state={
                  Object {
                    "capacity": "500",
                    "centerName": "Epic Center",
                    "cost": "50000",
                    "decision": "",
                    "description": "A center with emotions",
                    "errors": "",
                    "event": "",
                    "facilities": "car park",
                    "id": "",
                    "image": "newpicture.jpg",
                    "imageUrl": "picture.jpg",
                    "location": "Lekki",
                  }
                }
              />,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="col-lg-6 card bb text-center pt-4 pb-4"
            >
              <Details
                centerState={
                  Object {
                    "capacity": "500",
                    "centerName": "Epic Center",
                    "cost": "50000",
                    "decision": "",
                    "description": "A center with emotions",
                    "errors": "",
                    "event": "",
                    "facilities": "car park",
                    "id": "",
                    "image": "newpicture.jpg",
                    "imageUrl": "picture.jpg",
                    "location": "Lekki",
                  }
                }
                showHiddenDiv={[Function]}
              />
              <EditCenter
                centerState={
                  Object {
                    "capacity": "500",
                    "centerName": "Epic Center",
                    "cost": "50000",
                    "decision": "",
                    "description": "A center with emotions",
                    "errors": "",
                    "event": "",
                    "facilities": "car park",
                    "id": "",
                    "image": "newpicture.jpg",
                    "imageUrl": "picture.jpg",
                    "location": "Lekki",
                  }
                }
                imageUrl="newpicture.jpg"
                isValid={[Function]}
                onChange={[Function]}
                path="/"
                showHiddenDiv={[Function]}
                showImage={[Function]}
              />
            </div>
            <EventsList
              eventState={undefined}
              onClick={[Function]}
            />
            <DeleteModal
              content={
                <DeleteContent
                  onApprove={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  showHiddenDiv={[Function]}
                  state={
                    Object {
                      "capacity": "500",
                      "centerName": "Epic Center",
                      "cost": "50000",
                      "decision": "",
                      "description": "A center with emotions",
                      "errors": "",
                      "event": "",
                      "facilities": "car park",
                      "id": "",
                      "image": "newpicture.jpg",
                      "imageUrl": "picture.jpg",
                      "location": "Lekki",
                    }
                  }
                />
              }
              path="/"
            />
          </div>
        </div>,
        "id": "center-event",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
          >
            <div
              className="col-lg-6 card bb text-center pt-4 pb-4"
            >
              <Details
                centerState={
                  Object {
                    "capacity": "500",
                    "centerName": "Epic Center",
                    "cost": "50000",
                    "decision": "",
                    "description": "A center with emotions",
                    "errors": "",
                    "event": "",
                    "facilities": "car park",
                    "id": "",
                    "image": "newpicture.jpg",
                    "imageUrl": "picture.jpg",
                    "location": "Lekki",
                  }
                }
                showHiddenDiv={[Function]}
              />
              <EditCenter
                centerState={
                  Object {
                    "capacity": "500",
                    "centerName": "Epic Center",
                    "cost": "50000",
                    "decision": "",
                    "description": "A center with emotions",
                    "errors": "",
                    "event": "",
                    "facilities": "car park",
                    "id": "",
                    "image": "newpicture.jpg",
                    "imageUrl": "picture.jpg",
                    "location": "Lekki",
                  }
                }
                imageUrl="newpicture.jpg"
                isValid={[Function]}
                onChange={[Function]}
                path="/"
                showHiddenDiv={[Function]}
                showImage={[Function]}
              />
            </div>
            <EventsList
              eventState={undefined}
              onClick={[Function]}
            />
            <DeleteModal
              content={
                <DeleteContent
                  onApprove={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  showHiddenDiv={[Function]}
                  state={
                    Object {
                      "capacity": "500",
                      "centerName": "Epic Center",
                      "cost": "50000",
                      "decision": "",
                      "description": "A center with emotions",
                      "errors": "",
                      "event": "",
                      "facilities": "car park",
                      "id": "",
                      "image": "newpicture.jpg",
                      "imageUrl": "picture.jpg",
                      "location": "Lekki",
                    }
                  }
                />
              }
              path="/"
            />
          </div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="col-lg-6 card bb text-center pt-4 pb-4"
              >
                <Details
                  centerState={
                    Object {
                      "capacity": "500",
                      "centerName": "Epic Center",
                      "cost": "50000",
                      "decision": "",
                      "description": "A center with emotions",
                      "errors": "",
                      "event": "",
                      "facilities": "car park",
                      "id": "",
                      "image": "newpicture.jpg",
                      "imageUrl": "picture.jpg",
                      "location": "Lekki",
                    }
                  }
                  showHiddenDiv={[Function]}
                />
                <EditCenter
                  centerState={
                    Object {
                      "capacity": "500",
                      "centerName": "Epic Center",
                      "cost": "50000",
                      "decision": "",
                      "description": "A center with emotions",
                      "errors": "",
                      "event": "",
                      "facilities": "car park",
                      "id": "",
                      "image": "newpicture.jpg",
                      "imageUrl": "picture.jpg",
                      "location": "Lekki",
                    }
                  }
                  imageUrl="newpicture.jpg"
                  isValid={[Function]}
                  onChange={[Function]}
                  path="/"
                  showHiddenDiv={[Function]}
                  showImage={[Function]}
                />
              </div>,
              <EventsList
                eventState={undefined}
                onClick={[Function]}
              />,
              <DeleteModal
                content={
                  <DeleteContent
                    onApprove={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    showHiddenDiv={[Function]}
                    state={
                      Object {
                        "capacity": "500",
                        "centerName": "Epic Center",
                        "cost": "50000",
                        "decision": "",
                        "description": "A center with emotions",
                        "errors": "",
                        "event": "",
                        "facilities": "car park",
                        "id": "",
                        "image": "newpicture.jpg",
                        "imageUrl": "picture.jpg",
                        "location": "Lekki",
                      }
                    }
                  />
                }
                path="/"
              />,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Details
                    centerState={
                      Object {
                        "capacity": "500",
                        "centerName": "Epic Center",
                        "cost": "50000",
                        "decision": "",
                        "description": "A center with emotions",
                        "errors": "",
                        "event": "",
                        "facilities": "car park",
                        "id": "",
                        "image": "newpicture.jpg",
                        "imageUrl": "picture.jpg",
                        "location": "Lekki",
                      }
                    }
                    showHiddenDiv={[Function]}
                  />,
                  <EditCenter
                    centerState={
                      Object {
                        "capacity": "500",
                        "centerName": "Epic Center",
                        "cost": "50000",
                        "decision": "",
                        "description": "A center with emotions",
                        "errors": "",
                        "event": "",
                        "facilities": "car park",
                        "id": "",
                        "image": "newpicture.jpg",
                        "imageUrl": "picture.jpg",
                        "location": "Lekki",
                      }
                    }
                    imageUrl="newpicture.jpg"
                    isValid={[Function]}
                    onChange={[Function]}
                    path="/"
                    showHiddenDiv={[Function]}
                    showImage={[Function]}
                  />,
                ],
                "className": "col-lg-6 card bb text-center pt-4 pb-4",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "centerState": Object {
                      "capacity": "500",
                      "centerName": "Epic Center",
                      "cost": "50000",
                      "decision": "",
                      "description": "A center with emotions",
                      "errors": "",
                      "event": "",
                      "facilities": "car park",
                      "id": "",
                      "image": "newpicture.jpg",
                      "imageUrl": "picture.jpg",
                      "location": "Lekki",
                    },
                    "showHiddenDiv": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "centerState": Object {
                      "capacity": "500",
                      "centerName": "Epic Center",
                      "cost": "50000",
                      "decision": "",
                      "description": "A center with emotions",
                      "errors": "",
                      "event": "",
                      "facilities": "car park",
                      "id": "",
                      "image": "newpicture.jpg",
                      "imageUrl": "picture.jpg",
                      "location": "Lekki",
                    },
                    "imageUrl": "newpicture.jpg",
                    "isValid": [Function],
                    "onChange": [Function],
                    "path": "/",
                    "showHiddenDiv": [Function],
                    "showImage": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "eventState": undefined,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "content": <DeleteContent
                  onApprove={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  showHiddenDiv={[Function]}
                  state={
                    Object {
                      "capacity": "500",
                      "centerName": "Epic Center",
                      "cost": "50000",
                      "decision": "",
                      "description": "A center with emotions",
                      "errors": "",
                      "event": "",
                      "facilities": "car park",
                      "id": "",
                      "image": "newpicture.jpg",
                      "imageUrl": "picture.jpg",
                      "location": "Lekki",
                    }
                  }
                />,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
